R/utils-plot.R

```diff
@@ -135,6 +136,90 @@ formatTimeSeriesForPlotting <- function(series, removeZeroNegativeFlag=NULL){
+#' @return A list of formated time series data frames
+formatTimeSeriesListForPlotting <- function(seriesList, excludeZeroNegativeFlag=NULL){
+  if(!is.null(seriesList) && length(seriesList) > 0){
+    dataFrameList <- lapply(seriesList, function(e){
+      series <- e[['points']]
+
+      if(!isEmptyOrBlank(excludeZeroNegativeFlag) && excludeZeroNegativeFlag){
+        series <- removeZeroNegative(series)
+      }
+
+      if(nrow(series) == 0){
+        return(NULL)
+      }
+      
+      series[['legend.name']] <- e[['legend.name']]
+      return(as.list(series))
+    })
+
+    dataFrameList <- dataFrameList[unname(unlist(lapply(dataFrameList, function(e) {return(!is.null(e))})))]
+    return(dataFrameList)
+  }
+
+  return(NULL)
+}
+
+#' Plot Time Series
+#' 
+#' @description Function that takes a GS plot object and the necessary time series data and
+#' formats the time series properly for plotting and then plots it.
+#' @param plot_object The GS Plot object to plot onto
+#' @param ts The Time Series to plot
+#' @param name The variable name to use for the time series (used for style and config matching)
+#' @param yLabel The label to put onto the Y-Axis for this time series
+#' @param timezone The timezone of the time series (used for calculating gaps)
+#' @param excludeZeroNegativeFlag Whether or not to remove zero and negative values from the ts
+#' @param configFunction The function to use for fetching the style and config data for this TS
+#' @param isDV Whether or not the plot is a daily value plot (default: FALSE)
+plotTimeSeries <- function(plot_object, ts, name, yLabel, timezone, excludeZeroNegativeFlag, configFunction, isDV=FALSE){
+  if(!is.null(ts) && anyDataExist(ts[['points']])){
+    series <- splitDataGapsTimeSeries(ts, name, timezone, excludeZeroNegativeFlag, isDV=isDV)
+    series <- formatTimeSeriesListForPlotting(series, excludeZeroNegativeFlag)
+    
+    for(i in seq_len(length(series))){
+      plot_object <- plotItem(plot_object, series[[i]], name, configFunction, yLabel, isDV)
+    }
+  }
+  
+  return(plot_object)
+}
+
+#' Plot Item
+#' 
+#' @description Function that takes a GS plot object and the necessary item data and
+#' then formats the item properly for plotting and plots it.
+#' @param plot_object The GS Plot object to plot onto
+#' @param item The item to plot
+#' @param name The variable name to use for the item (used for style and config matching)
+#' @param configFunction The function to use for fetching the style and config data for this TS
+#' @param yLabel The label to put onto the Y-Axis for this item (default: "")
+#' @param isDV Whether or not the plot is a daily value plot (defulat: FALSE)
+plotItem <- function(plot_object, item, name, configFunction, yLabel="", isDV=FALSE){
+  if(!is.null(item) && anyDataExist(item)){
+    legendName <- item['legend.name']
+    plotItem <- configFunction(item, name, yLabel)
```

Comment:
didn't know you could do this! I thought you'd have to to use `do.call(configFunction, ...)`